front/lib/api/assistant/global_agents.ts

```diff
@@ -1318,6 +1321,24 @@ export async function getGlobalAgents(
     }
   }
 
+  // add user's favorite status to the agents if needed
+  const user = auth.user();
+  if (user) {
+    const relations = await AgentUserRelation.findAll({
+      where: {
+        userId: user.id,
+        agentConfiguration: globalAgents.map((agent) => agent.sId),
+      },
+    });
+
+    for (const agent of globalAgents) {
```

Comment:
What if the agent was deprecated?

Reply:
Sorry for omitting that one

AFAICT, if an agent was deprecated and returned by `_getGlobalAgents` prior to this PR, nothing will change with this PR; it will still be returned, with its `userFavorite` field correctly set